encoding/varint.ts

```diff
@@ -34,22 +38,51 @@ const SHIFTN = 7n;
  * from the returned new `offset`.
  */
 export function decode(buf: Uint8Array, offset = 0): [bigint, number] {
-  for (
-    let i = offset,
-      len = Math.min(buf.length, offset + MaxVarIntLen64),
-      shift = 0,
-      decoded = 0n;
-    i < len;
-    i += 1, shift += SHIFT
-  ) {
-    const byte = buf[i];
-    decoded += BigInt((byte & REST) * Math.pow(2, shift));
-    if (!(byte & MSB) && decoded > MaxUInt64) {
-      throw new RangeError("overflow varint");
+  U64_VIEW[0] = 0n;
+  let intermediate = 0;
+  let position = 0;
+  let i = offset;
+
+  if (buf.length === 0) throw new RangeError("Cannot read empty buffer");
+
+  let byte = buf[i];
```

Comment:
```suggestion
  let byte;
```

Reply:
What's funny is depending on some variables, removing the assignment here could reduce the performance due to how v8 does inlining. Incredibly weird. I think this should be noted in a comment here.

Reply:
Which variables? Benchmarks on my machine actually showed a small bump in performance from this small change.

Reply:
Skye and I were discussing this months (?) ago when she originally wrote this. I don't remember, deferring to Skye here. Perhaps a new version of v8 made this optimization worthless 🤷.

Reply:
it seems like it makes no difference anymore in the benchmarks. I guess this was just a style thing last time lol